datalad_next/iter_collections/annexworktree.py

```diff
@@ -0,0 +1,208 @@
+                # this avoids string operations to append newlines to items
+                input=intersperse(
+                    b'\n',
+                    # use `GitWorktree*`-elements yielded by `iter_gitworktree`
+                    # to create an `AnnexWorktreeItem` or
+                    # `AnnexWorktreeFileSystemItem` object, which is stored in
+                    # `git_fileinfo_store`. Yield a string representation of the
+                    # path contained in the `GitWorktree*`-element yielded by
+                    # `iter_gitworktree`
+                    route_out(
+                        glsf,
+                        git_fileinfo_store,
+                        lambda git_worktree_item: (
+                                str(git_worktree_item.name).encode(),
+                                git_worktree_item
+                        )
+                    )
+                )
+            ) as gaf, \
+            iter_subproc(
+                # get the key properties JSON-lines style
+                ['git', '-C', str(path), 'annex', 'examinekey', '--json', '--batch'],
+                # use only non-empty keys as input to `git annex examinekey`.
+                input=route_out(
+                    itemize(gaf, sep=None, keep_ends=True),
+                    key_store,
+                    # do not process empty key lines. Non-empty key lines
+                    # are processed, but nothing needs to be stored because the
+                    # processing result includes the key itself.
+                    lambda key: (StoreOnly, None)
+                                 if key.strip() == b''
+                                 else (key, None)
+                )
+            ) as gek:
+
+        results = route_in(
+            # the following `route_in` yields processed keys for annexed
+            # files and `StoreOnly` for non-annexed files. Its
+            # cardinality is the same as the cardinality of
+            # `iter_gitworktree`, i.e. it produces data for each element
+            # yielded by `iter_gitworktree`.
+            route_in(
+                load_json(itemize(gek, sep=None)),
+                key_store,
+                # `processed` data is either `StoreOnly` or detailed
+                # annex key information. we just return `process_data` as
+                # result, because `join_annex_info` knows how to incorporate
+                # it into an `AnnexWorktree*`-object.
+                lambda processed_data, _: processed_data
+            ),
+            git_fileinfo_store,
+            join_annex_info,
+        )
+
+        if not fp:
+            yield from results
+        else:
+            # Ensure that `path` is a `Path` object
+            path = Path(path)
+            for item in results:
+                item_path = content_path(item, path, link_target)
+                if item_path:
+                    with (path / item_path).open('rb') as active_fp:
```

Comment:
This does not work in general, because it does not anticipate directory records, which can come from untracked content.

Reply:
Thanks for catching that. There is a bug in `content_path`, which should only return existing annex-object paths, a path to an existing file, or a symlink target.

Will merge the fix after you pushed your changes